main/REPL/sharingService.md

```diff
@@ -0,0 +1,140 @@
+
+ **tyg todo: This doesn't make sense. The map name would seem to validate whether or not someone
+ else is already using it? **
+
+At the moment, there is a bug filed for the behavior shown here. While it would seem it should be the same,
+the `my` from `command[1]` is **not** the same as `history[1]`. `my` is the promise for the `sharedMap`. `history[1]`
+is the resolved version of that promise. Once the unresolved promise is assigned to `my`, `my` doesn't update its value 
+when the promise resolves.
+
+Thus, when `validate()` is called with an argument of `my`, it returns an error as `my` is an unresolved promise.
+While `history[1]` initially shows that unresolved promise, the REPL waits for it to resolve and prints the result
+as `history[1]`. If you later use a `history[1]` reference, it refers to the promise's resolved state.
+
+This behavior actually is **not** the bug. It's considered a quirk of the REPL. The bug is that `validate()` doesn't
+wait for any promise arguments to resolve. This section will be updated when `validate()` is revised to wait for 
+its parameter to resolve before trying to validate it. 
+
+```js
+command[1] my = E(home.sharingService).createSharedMap("MyMap")
+history[1] [Alleged: presence o-134]{}
+command[2] E(home.sharingService).validate(my)
+history[2] Promise.reject("Error: Unrecognized sharedMap: (an object)")
+command[3] my~.getName()
+history[3] "MyMap"
+command[4] E(home.sharingService).validate(history[1])
+history[4] [Alleged: presence o-134]{}
+```
+
+## `E(home.sharingService).sharedMap.addEntry(key, value)`
+- `sharedMap`: `{ sharedMap }`
+- `key`: `{ String }`
+- `value`: `{ Object }`
+- Returns: **tyg todo: Not sure**
+
+Adds an entry to the specified shared map with the specified key and value.
+
+**tyg todo: Still not sure how to use from REPL. See below.** 
+
+## `E(home.sharingService).sharedMap.lookupKey(key)` 
+- `key`: `{ String }`
+- Returns: `{ Object }`
+
+Returns the value associated with the specified key in the `sharedMap`.
+
+**tyg todo: Still not sure how to use from REPL. See below.**
+
+**tyg todo: OK, here's what I got in the REPL:**
+```js
+command[16] E(home.sharingService).createSharedMap("BMap")
+history[16] [Alleged: presence o-123]{}
+command[17] foo = E(home.sharingService).grabSharedMap("BMap")
+history[17] [Alleged: presence o-123]{}
+command[18] E(home.sharingService).foo.addEntry("key", "value")
+history[18] exception: "TypeError: E(...).foo.addEntry is not a function"
+command[20] E(foo).addEntry("key", "value")
+history[20] 1
+command[21] E(foo).lookupKey("key")
+history[21] Promise.reject("TypeError: target has no method \"lookupKey\", has [addEntry,getName,lookup]")
+command[23] E(home.sharingService).[history17].addEntry("foo", "bar")
+history[23] working on eval(E(home.sharingService).[history17].addEntry("foo", "bar"))
+command[24] E(home.sharingService).[history17].lookupKey("foo")
```

Comment:
history[17] isn't a property of sharing service, it's a variable that messages can be sent to on its own: 
```
history[17]~.lookup("foo")
```

If you continue to have trouble, let's do a shared session over zoom, and I can talk you through it.